lib/logthrsource/tests/test_logthrfetcherdrv.c

```diff
@@ -131,7 +131,6 @@ start_test_threaded_fetcher(TestThreadedFetcherDriver *s)
 {
   cr_assert(log_pipe_init(&s->super.super.super.super.super));
   cr_assert(log_pipe_on_config_inited(&s->super.super.super.super.super));
-  app_config_changed();
```

Comment:
Optional:
just a question, shouldn't we check if everything is "alright" after the reinit?
Maybe already exist this kind of test, just I didn't find it.

Reply:
I don't understand this question.

Reply:
Do you mean that we should write a test case where a config reinit with `AH_CONFIG_CHANGED` is tested with threaded sources/destinations?

That would be a really useful test, which could validate a potential bugfix for #3920.

Reply:
Sorry for the long wait, I didn't notice the comment :'(

Yes. Config reinit is the basic element of syslog-ng, but do a lot of stuff. My opinion is that could be good if we test it's everything run correctly after a config reinit.

I didn't know there is a related bug. I'm not sure it is possible to make a test case that can reproduce that bug every time (this is also my opinion :D - and please prove me wrong :) )

But one thing sure, I don't want to block this PR, I think that bug is could be in a new PR.